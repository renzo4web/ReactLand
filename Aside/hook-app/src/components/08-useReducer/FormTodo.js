import { useState } from "react";

const FormTodo = ({ addTodo }) => {
  const [input, setInput] = useState("");

  const handleSubmit = (event) => {
    event.preventDefault();

    if (!input.trim()) return;

    addTodo({
      id: new Date().getTime(),
      todo: input.trim(),
      completed: false,
    });
    setInput("");
  };
  return (
    <form onSubmit={handleSubmit}>
      <label className="form-label" htmlFor="todo">
        Add new todo
        <input
          className="form-control"
          value={input}
          onChange={({ target }) => setInput(target.value)}
          type="text"
        />
      </label>
      <button
        className="btn btn-outline-success"
        type="submit"
      >
        Add
      </button>
    </form>
  );
};

export default FormTodo;
