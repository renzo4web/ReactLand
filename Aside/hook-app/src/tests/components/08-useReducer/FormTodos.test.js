import { shallow } from "enzyme";
import FormTodo from "../../../components/08-useReducer/FormTodo";

describe("<FormTodo /> Test", () => {
  const addTodo = jest.fn();
  const wrapper = shallow(<FormTodo addTodo={addTodo} />);

  let formSubmit = wrapper.find("form").prop("onSubmit");

  test("should render properly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("should NOT  call addTodo() ", () => {
    formSubmit({ preventDefault() {} });
    expect(addTodo).toHaveBeenCalledTimes(0);
  });

  test("should call addTodo when form is submited", () => {
    const value = "Learn React";
    wrapper.find("input.form-control").simulate("change", {
      target: {
        value,
      },
    });

    formSubmit = wrapper.find("form").prop("onSubmit");
    formSubmit({ preventDefault() {} });

    expect(addTodo).toHaveBeenCalled();
    expect(addTodo).toHaveBeenCalledWith({
      todo: value,
      completed: false,
      id: expect.any(Number),
    });
  });
});
