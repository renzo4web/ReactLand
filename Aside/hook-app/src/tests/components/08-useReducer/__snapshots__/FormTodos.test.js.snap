// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormTodo /> Test should render properly 1`] = `
<form
  onSubmit={[Function]}
>
  <label
    className="form-label"
    htmlFor="todo"
  >
    Add new todo
    <input
      className="form-control"
      onChange={[Function]}
      type="text"
      value=""
    />
  </label>
  <button
    className="btn btn-outline-success"
    type="submit"
  >
    Add
  </button>
</form>
`;
